### PR TITLE
[KLC-2126] refactor(transaction): performance and correctness improvements

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary
- Optimize transaction processing hot paths with measurable performance gains
- Fix several correctness issues in permission validation, fee summation, and nil safety

## Problem
- `loadSignerPublicKeys` used a goroutine-per-signer pattern for a trivially cheap operation (`PublicKeyFromByteArray`), adding unnecessary scheduling overhead
- `validateTransactionSize` used `proto.Clone()` for temporary mutation, causing excessive allocations on every transaction validation
- `validatePermission` mixed error handling with threshold logic, fee summation lacked overflow protection, and `NewTxProcessor` had no nil check for `ForkController`

## Solution
- Replace goroutine-per-signer with sequential loop — benchmarked **6-8x faster** across all signer counts
- Replace `proto.Clone()` with defer-guarded save-mutate-restore — benchmarked **2.2x faster** with **7x fewer allocations**, remaining panic-safe
- Separate error handling from threshold check in `validatePermission`, add overflow protection in fee summation, add `ForkController` nil check, fix mismatched log keys and comment typos

## Key Changes
- **Updated:** `baseProcess.go` — sequential signer loading, save-restore pattern for size validation
- **Updated:** `interceptedTransaction.go` — refactored `validatePermission` with clearer error/threshold separation
- **Updated:** `txProcess.go` — nil check for `ForkController` in constructor
- **Updated:** `simulateTxProcess.go`, `txResultValidation.go` — minor log key and comment fixes
- **New:** `baseProcess_bench_test.go` — benchmarks for `loadSignerPublicKeys` and `validateTransactionSize`
- **Updated:** `baseProcess_test.go` — added threshold validation test coverage
- **New:** `export_test.go` — exported test helpers for benchmark access

## Testing
- [x] All existing tests pass (`make tests-unit`)
- [x] New benchmark tests validate performance claims
- [x] New unit tests for threshold edge cases

## Breaking Changes
None

## Configuration Changes
None

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by nickgs1337** [comment]
Please, open a Jira card explaining the source of this task, and link to this PR. Just so we don't lose track of modifications like this.

**Comment by fbsobreira** [comment]
> Please, open a Jira card explaining the source of this task, and link to this PR. Just so we don't lose track of modifications like this.

done!

File: core/process/transaction/baseProcess.go [hunk 1]
```diff
@@ -204,8 +187,13 @@ func (txProc *baseTxProcessor) checkTxValues(tx *transaction.Transaction, acntSn
...
 
 func (txProc *baseTxProcessor) CheckPaymentFeeBalance(tx *transaction.Transaction, acntSnd state.UserAccountHandler) error {
-	// check balance and fee
-	totalFees := tx.GetBandwidthFee() + tx.GetKAppFee()
+	// check balance and fee with overflow protection
+	bandwidthFee := tx.GetBandwidthFee()
+	kappFee := tx.GetKAppFee()
+	if bandwidthFee > 0 && kappFee > math.MaxInt64-bandwidthFee {
+		return fmt.Errorf("%w: fee calculation overflow", process.ErrOverflow)
+	}
```
Comment (Beroni) [inline core/process/transaction/baseProcess.go:195]: i think a new vailidation in baseTxProcessor need to be in a fork

**Review by Beroni — APPROVED** [review]

**Review by nickgs1337 — APPROVED** [review]
